site/src/components/ui/badge.tsx

```diff
@@ -0,0 +1,36 @@
+import { type VariantProps, cva } from "class-variance-authority";
+import type * as React from "react";
+
+import { cn } from "utils/cn";
+
+const badgeVariants = cva(
+	"inline-flex items-center rounded-md border px-2.5 py-1 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
+	{
+		variants: {
+			variant: {
+				default:
+					"border-transparent bg-surface-secondary text-content-secondary shadow hover:bg-surface-tertiary",
+				secondary:
+					"border-transparent bg-surface-secondary text-content-secondary hover:bg-surface-tertiary",
+				destructive:
+					"border-transparent bg-surface-error text-content-danger shadow hover:bg-surface-error/80",
+				outline: "text-content-primary",
+			},
+		},
+		defaultVariants: {
+			variant: "default",
+		},
+	},
+);
+
+export interface BadgeProps
+	extends React.HTMLAttributes<HTMLDivElement>,
+		VariantProps<typeof badgeVariants> {}
+
+function Badge({ className, variant, ...props }: BadgeProps) {
+	return (
+		<div className={cn(badgeVariants({ variant }), className)} {...props} />
+	);
+}
+
+export { Badge, badgeVariants };
```

Comment:
this is definitely another shadcn-ism we should avoid. just put `export` on the declaration if we need to. (I'm also not sure what value exporting a constant like this provides.)

Reply:
My original thought was to keep the shadcn components as close as possible to how they are imported but it makes sense to change this since the current codebase convention is not to group all exports at the bottom of the file. I think this pattern is generally used so its easy to look up all exports in one place. 

@Parkreiner  Since I have seen the export at the bottom pattern more often in other codebases, I am curious if you know why the current pattern exists?

Reply:
To be honest, I haven't been in enough "big" codebases to notice any "industry-leading" patterns, but Coder was just like this when I started working for them. Though I am a bigger fan of just exporting any relevant values directly

I know that `.d.ts` files often have all their exports at the bottom, but I don't know if that's a meaningful reference, since those are auto-generated anyway